File: study/ds/stack/stack.js
// Logic
class Stack {
  constructor() {
    this.stack = [];
  }

  push(item) {
    this.stack.push(item);
  }

  pop() {
    return this.stack.pop();
  }

  count() {
    return this.stack.length;
  }

  contains(item) {
    return this.stack.some((s) => s === item);
  }

  isEmpty() {
    return this.stack.length === 0;
  }
}

// Text
const stack = new Stack();
stack.push(1);
stack.push(2);
stack.push(3);
console.log(stack.count());
console.log(stack.pop());
console.log(stack.count());
console.log(stack.contains(2));
console.log(stack.contains(3));

File: study/ds/stack/stack.py
# https://www.javatpoint.com/data-structure-stack
# Logic
class Stack:
  def __init__(self):
    self.stack = []

  def push(self, item):
    self.stack.append(item)

  def pop(self):
    return self.stack.pop()
    
  def count(self):
    return len(self.stack)

  def contains(self, item):
    if item in self.stack:
      return True
    return False
  
  def isEmpty(self, item):
    return len(self.stack) == 0

# Test
stack = Stack();
stack.push(1)
stack.push(2)
stack.push(3)
print(stack.count())
print(stack.pop())
print(stack.count())
